src/data/finsupp/lattice.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson
+-/
+import data.finsupp
+import tactic.squeeze
+import algebra.ordered_group
+--open finsupp
+
+/-!
+# Lattice structure on finsupps
+
+This file provides instances of ordered structures on finsupps.
+
+-/
+
+open_locale classical
+noncomputable theory
+variables {α : Type*} {β : Type*} [has_zero β] {μ : Type*} [canonically_ordered_add_monoid μ]
+variables {γ : Type*} [canonically_linear_ordered_add_monoid γ]
+
+namespace finsupp
+
+lemma le_def [partial_order β] {a b : α →₀ β} : a ≤ b ↔ ∀ (s : α), a s ≤ b s := by refl
+
+instance : order_bot (α →₀ μ) :=
+{ bot := 0, bot_le := by simp [finsupp.le_def, ← bot_eq_zero], .. finsupp.partial_order}
+
+/-- Used to construct binary operations on `finsupp`s -/
+def binary_op_pointwise {f : β → β → β} (h : f 0 0 = 0) (a b : α →₀ β) : α →₀ β :=
+{ support := ((a.support) ∪ (b.support)).filter (λ s, f (a s) (b s) ≠ 0),
+  to_fun := λ s, f (a s) (b s),
+  mem_support_to_fun := by {
+    simp only [mem_support_iff, ne.def, finset.mem_union, finset.mem_filter],
+    intro s, apply and_iff_right_of_imp, rw ← not_and_distrib, rw not_imp_not,
+    intro h1, rw h1.left, rw h1.right, apply h, }, }
+
+lemma binary_op_pointwise_apply {f : β → β → β} (h : f 0 0 = 0) (a b : α →₀ β) (s : α) :
+  (binary_op_pointwise h a b) s = f (a s) (b s) := rfl
+
+instance [semilattice_inf β] : semilattice_inf (α →₀ β) :=
+{ inf := binary_op_pointwise inf_idem,
+  inf_le_left := by { intros, rw le_def, intro, apply inf_le_left, },
+  inf_le_right := by { intros, rw le_def, intro, apply inf_le_right, },
+  le_inf := by { intros, rw le_def at *, intro, apply le_inf (a_1 s) (a_2 s) },
+  ..finsupp.partial_order,
+}
+
+@[simp]
+lemma inf_apply [semilattice_inf β] {a : α} {f g : α →₀ β} : (f ⊓ g) a = f a ⊓ g a := rfl
+
+lemma support_inf {f g : α →₀ γ} : (f ⊓ g).support = f.support ∩ g.support :=
```

Comment:
Either both of neither of `support_inf`, `support_sup` should be simp lemmas. (Probably both.)